TASK [DDA-18] Create/Upload File to Aws

AndyLow14 opened the pull request:
# [DDA-18] Pull Request

## Description

- Get all tags on startup to display all available tags on the dropdown
- Select tags on dropdown and save to state
- Upload to AWS s3 and obtain download link
- Post both selected tagIds and awsUrl to the server
- Yarn seed to generate tags

## Type of change

Please delete options that are not relevant.

- [ ] RELEASE: Releasing a version
- [ ] HOTFIX: fix feature that would cause existing functionality to not work as expected
- [x] FEATURE: non-breaking change which adds functionality
- [ ] TASK: non-breaking change which adds functionality
- [ ] SUBTASK: non-breaking change which adds functionality

## For Developers

### Testing checklist

- [x] I have done manual testing on my code
  - [x] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [x] Apollo Studio
  - [ ] Postman
  - [x] Terminal
- [x] I have ensured existing unit tests pass locally with my changes
- [x] I have written new unit test

### Checklist

- [x] I have merged the target branch into the PR branch. Fix any conflicts that might appear
- [x] I have reviewed the code that I have written and ensured that there are no new warnings generated
- [x] I have followed the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation as required

## For Reviewers

### Testing checklist

- [ ] I have done manual testing on the code
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] Apollo Studio
  - [ ] Postman
  - [ ] Terminal
- [ ] I have ensured that all unit tests pass locally

### Checklist

- [ ] I have reviewed the code and ensured that there are no new warnings generated
- [ ] I have ensured the code follows the style guidelines of this project
- [ ] I have ensured that the required documentation has been updated


[DDA-18]: https://boutique-technologies.atlassian.net/browse/DDA-18?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AndyLow14 committed:
feat: get tags on run, select tags and post

- Get all tags on startup to display all available tags on the dropdown
- Select tags on dropdown and save to state
- Upload to AWS s3 and obtain download link
- Post both selected tagIds and awsUrl to the server

jnwh reviewed (changes requested):
1. Move inline styles to its own styles file and then use the Mantine create styles hooks.
2. Components are to be in their individually named folders. i.e. ModalForm.tsx should exist inside a folder named ModalForm.
3. Do not type something as "any", must have an explicit type. Only use "any" as a last resort. (Your typings can be within the component itself, but for more complex data structures it is better to put them in a types file inside the "types" folder)
4. "Magic number" values should be placed inside a constants file. i.e. Bucket Name, Region etc. (create a folder named constants to place your constants file)

AndyLow14 committed:
chore: files cleanup

- Moved inline styles to styles file
- Created folders for each component
- Removed any types
- Added @constants

AndyLow14 committed:
chore: restructured files, moved inline, fixed modal

AndyLow14 committed:
chore: edited style

jnwh reviewed (approved):
looks good shipit sendit
